### PR TITLE
SAScore module

### PR DESCRIPTION
**Pull request by lilleswing**
https://github.com/lilleswing/deepchem/blob/sascore/examples/notebooks/Synthetic_Feasibility_Scoring.ipynb
For a tutorial on how to use it.

This is also a prime example for storing pre-trained weights, probably should be the first case as the architecture is so simple.

### CODE REVIEW

File: deepchem/models/tensorgraph/layers.py [hunk 1]
```diff
@@ -4325,15 +4325,15 @@ def batch_mat_mult(self, A, B):
     return result
 
 
-class Hingeloss(Layer):
+class HingeLoss(Layer):
```
Comment (lilleswing) [inline deepchem/models/tensorgraph/layers.py]: Fix camel case issue in HingeLoss

File: deepchem/models/tensorgraph/tests/test_estimators.py [hunk 2]
```diff
@@ -277,3 +277,53 @@ def accuracy(labels, predictions, weights):
...
+    # Train the model.
+
+    estimator.train(input_fn=lambda: input_fn(100))
+
+    # Evaluate the model.
+
+    results = estimator.evaluate(input_fn=lambda: input_fn(1))
+    print(results)
+    assert results['loss'] < 1e-4
+    # TODO(LESWING) Discuss with peastman.
```
Comment (lilleswing) [inline deepchem/models/tensorgraph/tests/test_estimators.py]: @peastman I could use some help here.  

The model has 2 outputs -- scores 1-5 which are human interpretable levels of "complexity".

The loss function is a hinge-loss based on the difference between the two scores.

How does the Estimator calculate "accuracy"?

File: deepchem/molnet/load_function/tox21_datasets.py [hunk 3]
```diff
@@ -22,7 +22,7 @@ def load_tox21(featurizer='ECFP', split='index', reload=True, K=4):
 
   data_dir = deepchem.utils.get_data_dir()
   if reload:
-    save_dir = os.path.join(data_dir, "tox21/" + featurizer + "/" + split)
+    save_dir = os.path.join(data_dir, "tox21/" + featurizer + "/" + str(split))
```
Comment (lilleswing) [inline deepchem/molnet/load_function/tox21_datasets.py]: None couldn't be turned into a string.  This fix should go into all *_datasets.py files.

**Comment by lilleswing** [comment]
Can we use this as an example in our documentation for how to add a model from literature into dc.models.tensorgraph.models?

File: deepchem/models/tensorgraph/tests/test_estimators.py (hunk 2 above)
Comment (peastman) [reply deepchem/models/tensorgraph/tests/test_estimators.py]: You specify any metrics you're interested in when you call `make_estimator()`.  `evaluate()` returns the values of all of those metrics, as well as the loss.

Comment (lilleswing) [reply deepchem/models/tensorgraph/tests/test_estimators.py]: blerg feel dumb after reading the code.  That is obviously what is happening.  Updated swapping output[0] before calling create_metric in order to create a meaningful accuracy score.

File: deepchem/models/tensorgraph/models/test_sascore.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+import deepchem
+import numpy as np
+from deepchem.models import TensorGraph
+
+
+class TestSaScoreModel(unittest.TestCase):
+
+  def test_save_load(self):
+    """Test creating an Estimator from a ScScoreModel."""
```
Comment (abster12) [inline deepchem/models/tensorgraph/models/test_sascore.py]: The documentation should be a test for saving and loading the model. This applies for the estimator test.

**Comment by rbharath** [comment]
Wow this looks really good! Took a review pass and looks OK to merge from my end.
